public/app/features/query/components/QueryGroup.tsx

```diff
@@ -282,16 +278,11 @@ export class QueryGroup extends PureComponent<Props, State> {
     const { isDataSourceModalOpen } = this.state;
 
     const commonProps = {
-      enableFileUpload: config.featureToggles.editPanelCSVDragAndDrop,
-      fileUploadOptions: {
-        onDrop: this.onFileDrop,
-        maxSize: DFImport.maxFileSize,
-        multiple: false,
-        accept: DFImport.acceptedFiles,
-      },
       current: this.props.options.dataSource,
-      onChange: (ds: DataSourceInstanceSettings) => {
-        this.onChangeDataSource(ds);
+      queriesChanged: this.onQueriesChange.bind(this),
+      runQueries: this.props.onRunQueries,
```

Comment:
Why do we need these props to be added?  This API IMO is too hard to understand and too complex. Adding a CSV is just the same as selecting a DS, but also it gives you the query as well.

I see two possible APIs, that are simple enough to understand:

1. Torkel approach (https://github.com/grafana/grafana/issues/66914): receive the selected DS + queries
```jsx
const onChange = (ds: DataSourceInstanceSettings, queries: DataQuery[] | GrafanaQuery[]) => {
    this.onChangeDataSource(ds);
    this.onQueriesChange(queries);
    this.props.onRunQueries();
}:
<DataSourcePicker onChange={onChange}/>
```
* I like the simplicity and it will work for QueryGroup very well
* It is confusing to get queries from a DS Picker. We should find a good name for the param to be easier to understand
* Only when dropping a file you will receive the queries. This makes also the cb confusing

2. Have 2 callbacks: onChange, onUploadFile
```jsx
const onChange = (ds: DataSourceInstanceSettings) => {
    this.onChangeDataSource(ds);
    this.props.onRunQueries();
}:
const onUploadFile = (ds: DataSourceInstanceSettings, queries: GrafanaQuery[]) => {
    this.onChangeDataSource(ds);
    this.onQueriesChange(queries);
    this.props.onRunQueries();
};
<DataSourcePicker onChange={onChange} onUploadFile={onUploadFile} />
```
* It might be confusing why when dropping a file, the "onChange" is not called
* It gives a single purpose for each of the callbacks
* We could disable the D&D area if the callback is not provided or make it required if it is always enabled

I personally prefer the second approach. @torkelo any thoughts on this? 😄

Reply:
This does make sense, my first refactor was a bit rushed.

But looking at it now, is the onUploadFile cb really necessary? Clicking the Add CSV button also needs to add a query in order to set the QueryType to snapshot. Having the queries as an optional parameter to onChange would let us have just one cb for all three cases (normal change, file upload, Add CSV button).

Reply:
yea, there should by no means be any upload file or csv prop on DataSourcePicker not sure how that became a good idea :)   it select data sources, not anything else. Selecting a data source could however come with default queries